### PR TITLE
util: Add build workflow

### PR DESCRIPTION
**Pull request by benolayinka**
#### Summary

Create build workflow to lint and release docs as gh-pages

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Run Locally: Verified that the docs build using `make server`
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in `README.md`.
- [ ] Style Guidelines: Documentation obeys style guidelines in `README.md`.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Editor config
+      uses: snow-actions/eclint@v1.0.1
+      # TODO: Fix EditorConfig errors and remove
+      # https://github.com/TheThingsNetwork/lorawan-stack/issues/2723
+      continue-on-error: true
```
Comment (htdvisser) [inline .github/workflows/pull_request.yml:27]: This TODO points to an issue in the wrong repo

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        HUGO_PARAMS_INTERCOM_ENABLED: true
+        HUGO_PARAMS_SEARCH_APIKEY: ${{ secrets.HUGO_PARAMS_SEARCH_APIKEY }}
+        HUGO_PARAMS_SEARCH_ENABLED: true
+        HUGO_PARAMS_SEARCH_INDEX: thethingsstack
+    - name: Deploy documentation to Github pages
+      uses: JamesIves/github-pages-deploy-action@releases/v3
+      with:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+       BRANCH: gh-pages
+        FOLDER: doc/public
```
Comment (htdvisser) [inline .github/workflows/release.yml:36]: Indentation here doesn't look right

File: .editorconfig [hunk 3]
```diff
@@ -1,7 +1,7 @@
 root = true
 
 [*]
-max_line_length = 300
+#max_line_length = 300
```
Comment (htdvisser) [inline .editorconfig:4]: ?

File: doc/content/getting-started/aws/ecs/architecture/aws-databases.drawio [hunk 4]
```diff
@@ -1 +1 @@
...VQjSTU_BKqKpe2UHH" name="Page-1">7Vltb9owEP41fCzKGwQ+lgDbJCZ1Q1OlfqlM4iTWnDhznAL79TsnDpA4rJtGO7qFIuF7zjmf754zvjKwvWT3jqMs/sgCTAeWEewG9nxgWaZpjOFDInuFGIZdIREngcKOwJp8x/VEhRYkwHljomCMCpI1QZ+lKfZFA0Ocs21zWshoc9UMRVgD1j6iOnpPAhHXGxtPj4r3mESxWnpiuZUiQfVktZM8RgHbnkD2YmB7nDFRjZKdh6mMXh2X6rnlGe3BMY5T8SsPhIZnrb6QLEnn91Y49b89ONmNsvKEaKE2fPuECEUbQonYg+aBpVj5L/Z1UEJCqcco4yCmcoI9ywVnX3ENDizbdNzF7BY0AcpjLH0wQXjCXBCI7i0lUQqYYBmgIUvFWpk3lNxlSjkLNvDubBTMQ2yBlZglWHC5EfWAYwxHioH7GqnE7TG9I3M6dFQe45PkHuYixaroYP8YeBio2P9GHmwtD3ecPCGBZeCLTYqFloSMkVSUfoxm8AbPPGMwAo0npaE1agFt2W0Cpi5JG02gLbtNwGybN1vrm20HTwBNapg3WusbJw7C256xQlCSAgPrc0ASKeIoIPhIJkXWWCRUEXIbE4HXGfJlVLdwiNV0rE4i0+qgJ1RyJsfJLpKn3hBtc2cYcVZk5ZIf4Czq1D7m2C841NXjcfK6rJva8NkiOq04wBfjpbWcnK0npCSKQyHNwvZIGq1KaW7/rMIOxWpcptxco1Frrl5rztjQC82cvFShmSP9xCs442ggj88xlfHacBhFcnTHchFxvP600nUco+BmC+nUz8dOLrZCDn9L6bfGUdA5cxeUJ7o54WCIlLxKGZexa1MCXs5s2kWisHy1k1szZ4U2mMI+iTK/YUKw5Flq+eAV5s1aeq5uUJ5V4QjJTvrRXUgc56zgPq7KCMKcdxUUqpJ2EZLCpaTBUtPUaTrpYOnLkdTVSLqgKBfEQ36MO5n6GQck70l6XSTFZdL8MmkXYao1vjamTnum9kztumdbV8bU2p//qtWpm1GVBNu4hl7H0pvOvtnpm5233+yMp0PXaRac/dcbHvn8BRseltJ9/wX9tvudqc7Tjsb8db+d9X+A/ck9smfpP3KNtEZXyFWn52rP1Y77tvmaXAXx+NtRqTv5Cc5e/AA=</diagram></mxfile>
```
Comment (htdvisser) [inline doc/content/getting-started/aws/ecs/architecture/aws-databases.drawio:1]: If Draw.io doesn't automatically insert the newline, I think we should not enforce it.

So add to the `.editorconfig`:

```
[*.drawio]
insert_final_newline = false
```

File: go.mod [hunk 5]
```diff
@@ -3,7 +3,7 @@ module go.thethings.network/lorawan-stack-docs/v3
 go 1.14
 
 require (
-	github.com/TheThingsIndustries/magepkg v0.0.0-20190214092847-6c0299b7c3ed // indirect
-	github.com/gobuffalo/envy v1.9.0 // indirect
-	github.com/gohugoio/hugo v0.70.0
+  github.com/TheThingsIndustries/magepkg v0.0.0-20190214092847-6c0299b7c3ed // indirect
+  github.com/gobuffalo/envy v1.9.0 // indirect
+  github.com/gohugoio/hugo v0.70.0
```
Comment (htdvisser) [inline go.mod:8]: Formatting of `go.mod` is dictated by Go, so please fix `.editorconfig`.

**Review by htdvisser — CHANGES_REQUESTED** [review]

File: .editorconfig (hunk 3 above)
Comment (benolayinka) [reply .editorconfig:4]: we unfortunately don't obey this nearly anywhere. there were so many failures

Comment (johanstokking) [reply .editorconfig:4]: So, remove it, increase it, add a TODO comment?

For content, we shouldn't have this rule anyway. It's fine to write paragraphs on one line. In fact, I would strongly prefer that with text editing. I do think we need it for code files in this repository.

**Review by johanstokking — APPROVED** [review]

**Review by neoaggelos — COMMENTED** [review]
Nothing to add. Why are so many indent changes part of this PR?

**Review by htdvisser — APPROVED** [review]

**Review by kschiffer — APPROVED** [review]
Fine with me

**Comment by benolayinka** [comment]
> Nothing to add. Why are so many indent changes part of this PR?

PR adds `eclint` workflow job, they are from `eclint fix`

**Review by neoaggelos — APPROVED** [review]

**Pull request merged**